examples/gno.land/p/demo/forms/forms.gno

```diff
@@ -0,0 +1,131 @@
+package forms
+
+import (
+	"errors"
+	"std"
+	"time"
+
+	"gno.land/p/demo/seqid"
+)
+
+type Field struct {
+	Label string
+
+	/*
+		string: "string";
+		number: "number";
+		boolean: "boolean";
+		choice: "['Pizza', 'Schnitzel', 'Burger']";
+		multi-choice: "{'Pizza', 'Schnitzel', 'Burger'}";
+	*/
+	FieldType string
+
+	Required bool
+}
+
+type Form struct {
+	ID    string
+	Owner std.Address
+
+	Title       string
+	Description string
+	Fields      []Field
+
+	CreatedAt time.Time
+	openAt    *time.Time
+	closeAt   *time.Time
+}
+
+type Submission struct {
+	FormID string
+
+	Author std.Address
+
+	/* ["Alex", 21, true, 0, [0, 1]] */
+	Answers string // json
+
+	SubmittedAt time.Time
+}
+
+type FormDatabase struct {
+	Forms []*Form
+
+	Answers []*Submission
+
+	IDCounter seqid.ID
+}
+
+func NewDatabase() *FormDatabase {
+	return &FormDatabase{
+		Forms:   make([]*Form, 0),
+		Answers: make([]*Submission, 0),
+	}
+}
+
+func (form *Form) IsOpen() bool {
```

Comment:
I'd love to see at least a bit of godoc for all public functions, just to make them a bit more understandable at first glance

Reply:
i believe https://github.com/gnolang/gno/pull/2604/commits/361170ae0e657adfa4c841faf839c7bbf0343cc9 solves it but tell me if this is what you had in mind